pkg/cluster/cert.go

```diff
@@ -170,8 +173,9 @@ func GetSecret(ctx context.Context, c client.Client, namespace, name string) (*c
 func copySecretToNamespace(ctx context.Context, c client.Client, secret *corev1.Secret, newSecretName, namespace string) error {
 	newSecret := &corev1.Secret{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      newSecretName,
-			Namespace: namespace,
+			Name:            newSecretName,
+			Namespace:       namespace,
+			OwnerReferences: secret.OwnerReferences,
```

Comment:
This won't work if source and target secrets are in different namespaces. `OwnerReferences` can only be for local k8s resources.

Reply:
not sure i understand your comment here

the change is for this function to create a new Secret in the namespace with the Data and Type from the old secret also to set the new Secret with OwnerReferences as the origin FeatureTracker

Reply:
Unless I missed something, it looks like the new secret will have the same owner reference as the old secret with the line:
```
			OwnerReferences: secret.OwnerReferences,
```
Which, won't work if `secret` and `newSecret` are in different namespaces. For example, if the source secret is the default load balancer ingress cert from the ingress operator namespace being copied to another namespace.

Reply:
my idea is to set the `OwnerReferences: secret.OwnerReferences,` when we create the new secret.
but this value of `secret.OwnerReference` is not really the owner from old secret but set in the caller by 
```
        if err := ApplyMetaOptions(defaultIngressSecret, metaOptions...); err != nil {
		return err
	}
```
and passed from `feature.OwnedBy(f))`

In kserve's case: 
`ServingCertificateResource`() if it is using type "OpenshiftDefaultIngress"=> f *feature.Feature is added in metaOptions by `ApplyMetaOptions` => `copySecretToNamespace` use it as owner on new secret.

Reply:
some updates done  after above discussion : 

owner is only set within `copySecretToNamespace`() and passed from upper callers: `PropagateDefaultIngressCertificate`=>`copySecretToNamespace`=>`copySecretToNamespace`